augur/tasks/git/dependency_tasks/core.py

```diff
@@ -16,52 +20,133 @@ def generate_deps_data(session, repo_id, path):
         :param path: Absolute path of the Repostiory
         """
         
+        scan_date = datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
         session.logger.info('Searching for deps in repo')
-        session.logger.info(f'Repo ID: {repo_id}, Path: {path}')
-
-        deps = dep_calc.get_deps(path)
-        try: 
-            for dep in deps:
-                    repo_deps = {
-                        'repo_id': repo_id,
-                        'dep_name' : dep.name,
-    	                'dep_count' : dep.count,
-    	                'dep_language' : dep.language,
-                        'tool_source': 'deps_model',
-                        'tool_version': '0.43.9',
-                        'data_source': 'Git',
-                        'data_collection_date': datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
-                    }
-
-                    insert_statement = s.sql.text("""
-                        INSERT INTO "repo_dependencies" ("repo_id", "dep_name", "dep_count", "dep_language", "tool_source", "tool_version", "data_source", "data_collection_date")
-                        VALUES (:repo_id, :dep_name, :dep_count, :dep_language, :tool_source, :tool_version, :data_source, :data_collection_date)
-                    """).bindparams(**repo_deps)
-
-                    #result = self.db.execute(self.repo_dependencies_table.insert().values(repo_deps))
-                    session.execute_sql(insert_statement)
-        except Exception as e:
-            session.logger.error(f"Could not complete generate_deps_data!\n Reason: {e} \n Traceback: {''.join(traceback.format_exception(None, e, e.__traceback__))}")
-
-
-def deps_model(session, repo_id):
+        session.logger.info(f'Repo ID: {repo_id}, Path: {path}, Scan date: {scan_date}')
+
+        
+
+        deps = dep_calc.get_deps(path,session.logger)
+        
+        for dep in deps:
+            repo_deps = {
+                'repo_id': repo_id,
+                'dep_name' : dep.name,
+                'dep_count' : dep.count,
+                'dep_language' : dep.language,
+                'tool_source': 'deps_model',
+                'tool_version': '0.43.9',
+                'data_source': 'Git',
+                'data_collection_date': scan_date
+            }
+
+            session.insert_data(repo_deps,RepoDependency,["repo_id","dep_name","data_collection_date"])
```

Comment:
@IsaacMilarky I think this should insert all the of data at once so we aren't opening up a database transaction for every single row

Reply:
I agree.